woocommerce/woocommerce-ios
# Handle deleted Product Variations and Shipping Classes

Fixes #1636

## Summary

Similar to how we handled deleted Products by deleting all Products in the storage when syncing the first page of data, we do the same for Product Variations and Shipping Classes.

## Changes

- Centralized the default `firstPageNumber` to `Remote` and then `Store`'s `Default` enum
- In `PaginatedListSelectorViewController`, requested syncing the next page when the table view is about to display an index path. Also added some default configurations with estimated row heights that were missed before
- In `ProductShippingClassStore` and `ProductVariationStore`:
  - deleted the local models in the storage before upserting the sync results for the first page
  - actually passed in the page number and size to the remote call
- Added unit tests for `ProductShippingClassStore` and `ProductVariationStore` for syncing the first page and beyond

## Testing

Note: it might be easier to test paging by updating the `pageSize` in `SyncCoordinator` to a smaller number, but large enough so that the number of cells takes more than the screen height

Prerequisites:
- the store has more shipping classes than the `pageSize` for syncing
- a Product has more variations than the `pageSize` for syncing
---
- Launch the app
- Go to the Products tab
- Tap on a simple Product
- Tap "Edit" in the navigation bar
- Tap on the shipping settings row
- Tap on the shipping class row --> the paging should work as expected (the next page of data are loaded when scrolling to the bottom)
- On wp-admin, delete a shipping class
- Back in the app, pull down to refresh the shipping class list --> the deleted shipping class should not appear in the list
- Navigate back to the Shipping Settings and then tap the shipping class row --> the deleted shipping class should not appear in the list
- Go back to the Products tab
- Tap on the Product specified in the prerequisite with variations
- Tap on the Variations row --> the paging should work as expected (the next page of data are loaded when scrolling to the bottom)
- On wp-admin, delete a variation from the Product
- Back in the app, pull down to refresh the variation list --> the deleted variation should not appear in the list
- Navigate back to the Product Details and then tap the variations row --> the deleted shipping class should not appear in the list

Update release notes:
